docs/observability/metrics.md

```diff
@@ -0,0 +1,111 @@
+
+> [!Note]
+> The term "metric" can be overloaded and may refer to different concepts depending on the context. In Prometheus, counters, gauges, histograms, and summaries are all considered metric types. Histograms and summaries are more complex, representing collections of time series derived from multiple measurements. For clarity, we use "metric" here to refer to both raw measured values and their aggregated forms.
+
+### Micrometer in Camunda 8
+
+In Camunda 8, we use [Micrometer](https://micrometer.io/) as a facade for metrics collection (similar to how SLF4J works for logging). Micrometer provides:
+
+- Integration with observability systems like Prometheus, InfluxDB, Graphite, and others.
+- Out-of-the-box support for popular libraries and frameworks such as Spring Boot, Netty, and gRPC.
+
+### Visualization
+
+The visualization of metrics is done using [Grafana](https://grafana.com/), which can be connected to any of the supported observability systems.
+
+By default, we use [Prometheus](https://prometheus.io/) to collect, store, and alert on metrics.
+
+## Usage
+
+In Micrometer, a [MeterRegistry](https://docs.micrometer.io/micrometer/reference/concepts/registry.html) is the core component that tracks and manages all metrics (meters) in your application. Meters like counters, gauges, and timers are registered to a MeterRegistry, which then handles exporting these metrics to the configured monitoring systems.
+
+In Spring Boot, MeterRegistry instances are [auto-configured](https://docs.spring.io/spring-boot/docs/2.1.x/reference/html/production-ready-metrics.html#production-ready-metrics-getting-started) based on which observability system dependencies are present (e.g., Prometheus, InfluxDB).
+
+In the dist module, we typically use Autowiring to inject the registry where it's needed. In other cases, we pass it explicitly via constructor injection.
+
+This approach helps avoid spreading Spring dependencies throughout the codebase and improves testability by allowing easier mocking or substitution of the registry in tests.
+
+Example:
+
+```java
+final MeterRegistry meterRegistry = ...; // injected
+final  var counter = Counter.builder("my.cool.counter")
+  .description("A super cool counter")
+  .register(meterRegistry);
+counter.increment(); // increments the counter by 1
+```
+
+## Adding a metric
+
+1. Get access to the meter registry
+
+[//]: # (ask Stephan to explain the issue with setting up meter registry)
+
+2. Pick a metric type
+
+- If you need to observe a value change over time, use a `Counter`
+
+  > [!Note] Counters can only increment (they cannot be decremented). They reset when the application restarts, but time-derivative functions like Prometheus's rate() or increase() can accommodate this reset when used with time series data.
+
+- If you need to observe the current value at a specific moment in time, use a `Gauge`. Gauges can go up or down and are useful for things like memory usage, thread counts, or queue sizes
```

Comment:
for the next iteration of this guide 😅 (just dropping some thought here)
- Gauges are very special as they only provide the current snapshot of the metric (you loose information in between snapshots)
- prometheus and other systems poll data from the nodes/servers/apps in defined intervals (I think this would also be worth mentioning)
- that means when looking at this metric in grafana we should keep in mind, that we don't know the gauge value in between polls, e.g. we don't know if it went up 500 and down 400 or just went up 100